airbyte-integrations/bases/connector-acceptance-test/unit_tests/test_core.py

```diff
@@ -1331,6 +1337,188 @@ def test_validate_field_appears_at_least_once(records, configured_catalog, expec
+                AirbyteMessage(
+                    type=Type.TRACE,
+                    trace=AirbyteTraceMessage(
+                        type=TraceType.STREAM_STATUS,
+                        emitted_at=120,
+                        stream_status=AirbyteStreamStatusTraceMessage(
+                            stream_descriptor=StreamDescriptor(name="test_stream_0"), status=AirbyteStreamStatus.COMPLETE
+                        ),
+                    ),
+                ),
+            ],
+            does_not_raise(),
+        ),
+        (
+            [
+                AirbyteMessage(type=Type.RECORD, record=AirbyteRecordMessage(stream="test_stream_0", data={"a": 1}, emitted_at=111)),
+                AirbyteMessage(type=Type.RECORD, record=AirbyteRecordMessage(stream="test_stream_1", data={"a": 1}, emitted_at=112)),
+                AirbyteMessage(type=Type.RECORD, record=AirbyteRecordMessage(stream="test_stream_2", data={"a": 1}, emitted_at=113)),
+            ],
+            pytest.raises(AssertionError),
+        ),
+        (
+            [
+                AirbyteMessage(
+                    type=Type.TRACE,
+                    trace=AirbyteTraceMessage(
+                        type=TraceType.STREAM_STATUS,
+                        emitted_at=1,
+                        stream_status=AirbyteStreamStatusTraceMessage(
+                            stream_descriptor=StreamDescriptor(name="test_stream_0"), status=AirbyteStreamStatus.STARTED
+                        ),
+                    ),
+                ),
+                AirbyteMessage(
+                    type=Type.TRACE,
+                    trace=AirbyteTraceMessage(
+                        type=TraceType.STREAM_STATUS,
+                        emitted_at=1,
+                        stream_status=AirbyteStreamStatusTraceMessage(
+                            stream_descriptor=StreamDescriptor(name="test_stream_2"), status=AirbyteStreamStatus.STARTED
+                        ),
+                    ),
+                ),
+                AirbyteMessage(
+                    type=Type.TRACE,
+                    trace=AirbyteTraceMessage(
+                        type=TraceType.STREAM_STATUS,
+                        emitted_at=1,
+                        stream_status=AirbyteStreamStatusTraceMessage(
+                            stream_descriptor=StreamDescriptor(name="test_stream_1"), status=AirbyteStreamStatus.STARTED
+                        ),
+                    ),
+                ),
+                AirbyteMessage(type=Type.RECORD, record=AirbyteRecordMessage(stream="test_stream_0", data={"a": 1}, emitted_at=111)),
+                AirbyteMessage(type=Type.RECORD, record=AirbyteRecordMessage(stream="test_stream_1", data={"a": 1}, emitted_at=112)),
+                AirbyteMessage(type=Type.RECORD, record=AirbyteRecordMessage(stream="test_stream_2", data={"a": 1}, emitted_at=113)),
+            ],
+            pytest.raises(AssertionError),
+        ),
```

Comment:
There are interesting cases missing:
* STARTED, RUNNING but not COMPLETE
* STARTED, RUNNING, RUNNING, COMPLETE
* STARTED, RUNNING and INCOMPLETE
* Just COMPLETE

Reply:
add coverage for this cases.
INCOMPLETE status cannot be tested as of now, reason: https://github.com/airbytehq/airbyte/issues/33565#issuecomment-1919024335